tests/level-db/test_leveldb_db_backend.py

```diff
@@ -0,0 +1,42 @@
+import os
+import pytest
+from evm.db.backends.memory import MemoryDB
+from evm.db import (
+    get_db_backend,
+)
+
+
+pytest.importorskip('leveldb') 
+memoryDb = MemoryDB()
+
+
+def test_set():
+    # Sets ecc backend to leveldb
+    os.environ['EVM_DB_BACKEND_CLASS'] = 'evm.db.backends.level.LevelDB'
+    assert get_db_backend().set(b'1', b'1') == memoryDb.set(b'1', b'1') 
+
+
+def test_get():
+    assert get_db_backend().get(b'1') == memoryDb.get(b'1') 
+
+
+def test_set_on_existing_value():
+    get_db_backend().set(b'1', b'2') == memoryDb.set(b'1', b'2') 
+    assert get_db_backend().get(b'1') == memoryDb.get(b'1')
+
+def test_exists():
+    assert get_db_backend().exists(b'1') == memoryDb.exists(b'1')
+
+def test_delete():
+    assert get_db_backend().delete(b'1') == memoryDb.delete(b'1')
+    assert get_db_backend().exists(b'1') == memoryDb.exists(b'1')
+
+def test_snapshot_and_revert():
+    snapshot = get_db_backend().snapshot()
+    get_db_backend().set(b'1', b'1')
+    assert get_db_backend().get(b'1')
+    with pytest.raises(KeyError): snapshot.get(b'1')
+    get_db_backend().revert(snapshot)
+    with pytest.raises(KeyError): get_db_backend().get(b'1')
+    # Switches db backend back to local memory for the rest of the tests
+    os.environ['EVM_DB_BACKEND_CLASS'] = 'evm.db.backends.memory.MemoryDB'
```

Comment:
Look up the `monkeypatch` fixture provided by pytest.  It has an API for setting environment variables in asuch a way that they are reset after the test is run.